rbc/omnisci_array.py

```diff
@@ -167,12 +167,66 @@ def omnisci_array_setitem(a, i, v):
         return lambda a, i, v: omnisci_array_setitem_(a, i, v)
 
 
-@extending.overload(np.sum)
+@extending.intrinsic
+def omnisci_array_logical_and_(typingctx, a, b):
+    sig = types.boolean8(a, b)
+
+    def codegen(context, builder, signature, args):
+        breakpoint()
+
+    return sig, codegen
+
+
+@extending.overload_method(ArrayPointer, 'fill')
+def omnisci_array_fill(x, v):
+    if isinstance(x, ArrayPointer):
+        def impl(x, v):
+            for i in range(len(x)):
+                x[i] = v
+        return impl
+
+
+@extending.overload(max)
+@extending.overload(np.max)
+@extending.overload_method(ArrayPointer, 'max')
+def omnisci_array_max(x, initial=None):
+    if isinstance(x, ArrayPointer):
+        def impl(x, initial=None):
+            if initial is not None:
+                m = initial
+            else:  # XXX: check if len(array) > 0
```

Comment:
Add the check and raise `ValueError: zero-size array` when `len(x)==0`.

However, raising an exception will likely not work. If so, add `printf('omnisci_array_max: cannot find max of zero-sized array')` or similar, and return lowest value of `x` dtype (when float then return `-inf`, when int8 then return -127, etc).
This approach could be feasible only if we cannot handle exceptions via omniscdb at all and to avoid crashing the server, we need to return something sensible.

Reply:
Yes, raising an exception will not work. Numba serializes the exception as a constant string on LLVM and deserializes once it returns to the python interpreter. So, `raise ValueError(' zero-size array')` becomes `@.const.pickledata.26137249904 = internal constant [77 x i8] c"\80\04\95B\00\00\00\00\00\00\00\8C\08builtins\94\8C\0AValueError\94\93\94\8C\0Fzero-size array\94\85\94\8C\03foo\94\8C\04a.py\94K\06\87\94\87\94."` on llvm IR.

Reply:
Opened https://github.com/xnd-project/rbc/issues/80